aspnetcore/signalr/javascript-client.md

```diff
@@ -98,7 +98,140 @@ Use the [configureLogging](/javascript/api/%40aspnet/signalr/hubconnectionbuilde
+
+> [!WARNING]
+> The onreconnected callbacks connectionId parameter will not be undefined if the `HubConnection` was configured to [skip negotiation.](xref:signalr/configuration#configure-client-options)
+
+```javascript
+connection.onreconnected((connectionId) => {
+  console.assert(connection.sate === signalR.HubConnectionState.Connected);
+
+  document.getElementById("messageInput").disabled = false;
+
+  const li = document.createElement("li");
+  li.textContent = `Connection reestablished. Connected with connectionId "${connectionId}".`;
+  document.getElementById("messagesList").appendChild(li);
+});
+```
+
+`withAutomaticReconnect()` will not configure the HubConnection to retry initial start failures, so start failures need to be handled manually:
+
+```javascript
+async function start() {
+    try {
+        await connection.start();
+        console.assert(connection.sate === signalR.HubConnectionState.Connected);
+        console.log("connected");
+    } catch (err) {
+        console.assert(connection.sate === signalR.HubConnectionState.Disconnected);
+        console.log(err);
+        setTimeout(() => start(), 5000);
+    }
+};
+```
+
+If the client doesn't succesfully reconnect within its first four attempts, the `HubConnection` will transition to the [Disconnected](javascript/api/%40aspnet/signalr/hubconnectionstate#reconnecting) state and fire its [onclosed](/javascript/api/%40aspnet/signalr/hubconnection#onreconnected) callbacks. This is a good opportunity to inform users the connection has been permanently lost and recommend refreshing the page:
+
+```javascript
+connection.onclose((error) => {
+  console.assert(connection.sate === signalR.HubConnectionState.Disconnected);
+
+  document.getElementById("messageInput").disabled = true;
+
+  const li = document.createElement("li");
+  li.textContent = `Connection closed due to error "${error}". Try refreshing this page to restart the connection.`;
+  document.getElementById("messagesList").appendChild(li);
+})
+```
+
+In order to configure a non-defualt number of reconnect attempts before failure, or to change the reconnect timing, [withAutomaticReconnect](/javascript/api/%40aspnet/signalr/hubconnectionbuilder#withautomaticreconnect) accepts an array of numbers representing the delay in milliseconds to wait before starting each reconnect attempt.
+
+```javascript
+const connection = new signalR.HubConnectionBuilder()
+    .withUrl("/chatHub")
+    .withAutomaticReconnect([0, 0, 10000])
+    .build();
+
+    // .withAutomaticReconnect([0, 2000, 10000 and 30000]) yields the default behavior
+```
+
+The above example configures the `HubConnection` to start attempting reconnects immediately after the connection is lost. This is also true for the default configuration.
+
+If the first reconnect attempt fails, the second reconnect attempt will also start immediately instead of waiting 2 seconds like it would in the default configuration.
```

Comment:
Were the new lines intentional here?

Reply:
Yeah. I tried to put the description of each custom reconnect attempt as compared to the default configuration in its own paragraph/text block.